pymeasure/instruments/hp/hp66312A.py

```diff
@@ -0,0 +1,288 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+#work in progress by TEM01STAR
+from pymeasure.instruments import Instrument, SCPIUnknownMixin
+from pymeasure.instruments.validators import strict_discrete_set
+
+class HP66312A(SCPIUnknownMixin, Instrument):
```

Comment:
Does your device support the commands defined by the SCPIMixin (parent of the unknown mixin) ?
if so, please use SCPIMixin instead.
if not, please do not inherit the unknown mixin.

The unknown mixin is a crutch for us maintainers transitioning from default assuming Scpi compliance to explicitly defining it.
for older instruments we do not know, whether the device supports Scpi (having been assumed implicitly) or not (forgotten to define that it is not compliant).
no new instrument should inherit it.

Reply:
Thank you for the clarification!